napari/_qt/qt_main_window.py

```diff
@@ -1162,21 +1165,27 @@ def _screenshot(
                     )
                 prev_size = canvas.size
                 canvas.size = size[::-1]  # invert x ad y for vispy
+            if scale is not None:
+                prev_size = canvas.size
```

Comment:
This breaks when both `size` and `scale` are passed, because it will set `prev_size` to the already resized canvas; at the end, when you set `canvas.size = prev_size` it won't be right.

Reply:
Derp, you're absolutely right. I guess the options are:
1. pull `prev_size = canvas.size` out and ahead of both `ifs`. Simple, but will be set every time, whether needed or not.
2. add a check before both ifs: `if size is not None or scale is not None:` to set the `prev_size`. More code, but only used when needed.

Reply:
Github is having some issues...

Reply:
I committed Option1
https://github.com/brisvag/napari/pull/5/commits/adee3e092774e3557148d11a3de9f8dfdb8c25c3